Sources/FoundationEssentials/Subprocess/Subprocess+Configuration.swift

```diff
@@ -0,0 +1,615 @@
+                errorCloseFunc = process.executionError.closeParentSide
+            }
+
+            var inputError: Error?
+            var outputError: Error?
+            var errorError: Error? // lol
+            do {
+                try inputCloseFunc()
+            } catch {
+                inputError = error
+            }
+
+            do {
+                try outputCloseFunc()
+            } catch {
+                outputError = error
+            }
+
+            do {
+                try errorCloseFunc()
+            } catch {
+                errorError = error // lolol
+            }
+
+            // Attempt to kill the subprocess
+            var killError: Error?
+            if attemptToTerminateSubProcess {
+                do {
+                    try process.sendSignal(.kill, toProcessGroup: true)
+                } catch {
+                    guard let posixError: POSIXError = error as? POSIXError else {
+                        killError = error
+                        return
+                    }
+                    // Ignore ESRCH (no such process)
+                    if posixError.code != .ESRCH {
+                        killError = error
+                    }
+                }
+            }
+
+            if let inputError = inputError {
+                throw inputError
+            }
+
+            if let outputError = outputError {
+                throw outputError
+            }
+
+            if let errorError = errorError {
+                throw errorError
+            }
+
+            if let killError = killError {
+                throw killError
+            }
+        }
+
+        /// Close each input individually, and throw the first error if there's multiple errors thrown
+        @Sendable
+        internal func cleanupAll(
+            input: ExecutionInput,
+            output: ExecutionOutput,
+            error: ExecutionOutput
+        ) throws {
+            var inputError: Error?
+            var outputError: Error?
+            var errorError: Error?
+
+            do {
+                try input.closeAll()
+            } catch {
+                inputError = error
+            }
+
+            do {
+                try output.closeAll()
+            } catch {
+                outputError = error
+            }
+
+            do {
+                try error.closeAll()
+            } catch {
+                errorError = error
+            }
+
+            if let inputError = inputError {
+                throw inputError
+            }
+            if let outputError = outputError {
+                throw outputError
+            }
+            if let errorError = errorError {
+                throw errorError
+            }
+        }
+
+        internal func run<R>(
+            output: RedirectedOutputMethod,
+            error: RedirectedOutputMethod,
+            _ body: @Sendable @escaping (Subprocess, StandardInputWriter) async throws -> R
```

Comment:
The `body` really shouldn't require to be `@Sendable` nor `@escaping`. This will limit the way how this is used and make it very hard to use from within an actor.